doc/source/rllib/learner-connector.rst

```diff
@@ -0,0 +1,381 @@
+
+    class CountBasedIntrinsicRewards(ConnectorV2):
+
+        def __init__(self, **kwargs):
+            super().__init__(**kwargs)
+
+            # Observation counter to compute state visitation frequencies.
+            self._counts = Counter()
+
+
+In the :py:meth:`~ray.rllib.connectors.connector_v2.ConnectorV2.__call__` method, you then loop through all
+single-agent episodes and change the reward stored in these to: ``r(t) = re(t) + 1 / N(ot)``, where ``re`` is the extrinsic reward from the
+RL environment and ``N(ot)`` is the number of times the agent has already been to observation ``o(t)``.
+
+.. testcode::
+
+        def __call__(
+            self,
+            *,
+            rl_module,
+            batch,
+            episodes,
+            explore=None,
+            shared_data=None,
+            **kwargs,
+        ) -> Any:
+            for sa_episode in self.single_agent_episode_iterator(
+                episodes=episodes, agents_that_stepped_only=False
+            ):
+                # Loop through all observations, except the last one.
+                observations = sa_episode.get_observations(slice(None, -1))
+                # Get all respective extrinsic rewards.
+                rewards = sa_episode.get_rewards()
+
+                for i, (obs, rew) in enumerate(zip(observations, rewards)):
+                    # Add 1 to obs counter.
+                    obs = tuple(obs)
+                    self._counts[obs] += 1
+                    # Compute the count-based intrinsic reward and add it to the extrinsic
+                    # reward.
+                    rew += 1 / self._counts[obs]
+                    # Store the new reward back to the episode (under the correct
+                    # timestep/index).
+                    sa_episode.set_rewards(new_data=rew, at_indices=i)
+
+            return batch
+
+
+If you plug in this custom :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2` piece into the pipeline through
+the algorithm config
+(`config.learners(learner_connector=lambda env: CountBasedIntrinsicRewards())`),
+your loss function should receive the altered reward signals in the ``rewards`` column of the incoming batch.
+
+.. note::
+    Your custom logic writes the new rewards right back into the given episodes
+    instead of placing them into the train batch. This strategy of writing back those data you pulled from episodes right back
+    into the same episodes makes sure that from this point on, only the changed data is visible to the subsequent connector pieces.
+    The batch remains unchanged at first. However, one of the subsequent
+    :ref:`default Learner connector pieces <default-learner-pipeline>`, :py:class:`~ray.rllib.connectors.learner.add_columns_from_episodes_to_batch.AddColumnsFromEpisodesToBatch`,
+    fills the batch with rewards data from the episodes.
+    Thus, you can rely on the information you changed in the episode to automatically end up in the train batch in the end.
```

Comment:
"Thus, ..." -> "Therefore, any changes you make to the episode object will automatically be included in the final training batch."?

Reply:
Oh, also, we cannot use passive voice (same: vale) :)
So we should always formulate like:
"RLlib automatically then does ABC..." instead of "ABC is [some passive construction] automatically ..."